src/app/components_ngrx/label-selector/label-selector.component.ts

```diff
@@ -58,8 +58,10 @@ export class LabelSelectorComponent implements OnInit {
   }
 
   @Input('selectedLabels') set selectedLabelsSetter(labels: LabelUI[]) {
-    this.selectedLabels = labels;
-    this.updateSelection();
+    if (this.selectedLabels !== labels) {
```

Comment:
Before going into any context, how are you comparing to arrays ??? This will be false !!

Reply:
I am not going to (will use lodash `isEqual` for that), but good that you noticed.

This WIP first cut was meant to ask you about whether I'm making the change in right place/context (as the data flow in the component isn't super intuitive). The code itself can be considered pseudocode before I get the feedback.